packages/driver/src/cy/net-stubbing/add-command.ts

```diff
@@ -131,7 +131,7 @@ function validateRouteMatcherOptions (routeMatcher: RouteMatcherOptions): { isVa
     }
   }
 
-  if (_.isString(routeMatcher.hostname) && !isValidDomain(routeMatcher.hostname, { allowUnicode: true })) {
+  if (_.isString(routeMatcher.hostname) && !isValidDomain(routeMatcher.hostname, { allowUnicode: true }) && routeMatcher.hostname !== 'localhost') {
```

Comment:
This change is not comprehensive enough, I think any hostname should be allowed to work here. For example, `somemachine`, `local6`, `router`, etc. should all work.

You could try to patch `isValidDomain`: https://github.com/miguelmota/is-valid-domain/issues/23

Alternatively, you could kludge it here, by checking if `hostname` contains a `.`, and if not, appending `.sometld` before passing it to `isValidDomain`...

Reply:
or https://github.com/miguelmota/is-valid-hostname, of course